File: Sdks/AdminApi.Sdk/src/Endpoints/Tiers/Types/TierQuotaDefinition.cs
using Backbone.AdminApi.Sdk.Endpoints.Identities.Types;

namespace Backbone.AdminApi.Sdk.Endpoints.Tiers.Types;

public class TierQuotaDefinition
{
    public required string Id { get; set; }
    public required Metric Metric { get; set; }
    public required int Max { get; set; }
    public required string Period { get; set; }
}

File: Applications/AdminUi/packages/admin_api_types/lib/src/tiers/tier_quota_definition.dart
import 'package:json_annotation/json_annotation.dart';

import '../metrics/metric.dart';

part 'tier_quota_definition.g.dart';

@JsonSerializable()
class TierQuotaDefinition {
  final String id;
  final Metric metric;
  final int max;
  final String period;

  TierQuotaDefinition({required this.id, required this.metric, required this.max, required this.period});

  factory TierQuotaDefinition.fromJson(dynamic json) => _$TierQuotaDefinitionFromJson(json as Map<String, dynamic>);
  Map<String, dynamic> toJson() => _$TierQuotaDefinitionToJson(this);
}
